### PR TITLE
add header component

### PR DESCRIPTION
**Pull request by Mihir0000**

### CODE REVIEW

File: src/Components/Header/HeaderCarousel.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                                        src="/static/image/carousel-img-1.png"
+                                        alt="First slide"
+                                    />
+                                </div>
+                            </Carousel.Item>
+                            <Carousel.Item>
+                                <div className="d-flex flex-row">
+                                    <div className="w-100 d-flex justify-content-center align-item-center align-self-center">
+                                        <h2 className="txt col-8 justify-content-start text-white fs-2">
+                                            Easy Fix <br/> Superior Service Quality Parts
```
Comment (Reeju2019) [inline src/Components/Header/HeaderCarousel.tsx:54]: Refer to earlier review and change accordingly

File: src/Components/Header/HeaderCarousel.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
```
Comment (Reeju2019) [inline src/Components/Header/HeaderCarousel.tsx:1]: Please change this to `import React from 'react';`

File: src/Components/Header/HeaderCarousel.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import { Carousel, Container, Image } from 'react-bootstrap';
+import './Header.css';
+
+interface IHeaderCarouselProps {}
+
+const HeaderCarousel: React.FunctionComponent<IHeaderCarouselProps> = (
+    props
```
Comment (Reeju2019) [inline src/Components/Header/HeaderCarousel.tsx:8]: If their is no props then please remove this

**Review by Reeju2019 — CHANGES_REQUESTED** [review]
I am not sure if this code was ready for my review.
Please refactor this code and push the changes.

File: src/Components/Header/HeaderCarousel.tsx (hunk 3 above)
Comment (Mihir0000) [reply src/Components/Header/HeaderCarousel.tsx:8]: done

Comment (Mihir0000) [reply src/Components/Header/HeaderCarousel.tsx:8]: done

File: src/Components/Header/HeaderCarousel.tsx (hunk 2 above)
Comment (Mihir0000) [reply src/Components/Header/HeaderCarousel.tsx:1]: done

File: src/Components/Header/HeaderCarousel.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                                        src="/static/image/carousel-img-1.png"
+                                        alt="First slide"
+                                    />
+                                </div>
+                            </Carousel.Item>
+                            <Carousel.Item>
+                                <div className="d-flex flex-row">
+                                    <div className="w-100 d-flex justify-content-center align-item-center align-self-center">
+                                        <h2 className="txt col-8 justify-content-start text-white fs-2">
+                                            Top Trained Experts <br/> at Your Doorstep
```
[the remark this answers is not in the document]
Comment (Mihir0000) [reply src/Components/Header/HeaderCarousel.tsx:39]: done

File: src/Components/Header/HeaderCarousel.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                        id="carouselExampleIndicators"
+                        className="carousel slide"
+                        data-bs-ride="true"
+                    >
+                        <Carousel>
+                            <Carousel.Item>
+                                <div className="d-flex flex-row">
+                                    <div className="w-100 d-flex justify-content-center align-item-center align-self-center">
+                                        <h2 className="txt col-8 justify-content-start text-white fs-2">
+                                            Repair & Maintainence <br />
```
[the remark this answers is not in the document]
Comment (Mihir0000) [reply src/Components/Header/HeaderCarousel.tsx:24]: done

File: src/Components/Header/HeaderCarousel.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { Carousel, Container, Image } from 'react-bootstrap';
+import './Header.css';
+
+interface IHeaderCarouselProps {}
```
[the remark this answers is not in the document]
Comment (Mihir0000) [reply src/Components/Header/HeaderCarousel.tsx:5]: done

File: src/Components/Header/HeaderCarousel.tsx (hunk 1 above)
Comment (Mihir0000) [reply src/Components/Header/HeaderCarousel.tsx:54]: done

File: src/Components/Header/Header.css [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  border-left: 0!important;
+}
+
+.caro-img{
+  width: 25vw;
+  
+}
+
+.carousel-container-1 {
+    background: url(../../../public/static/image/carousel-bg-1.png);
```
Comment (Reeju2019) [inline src/Components/Header/Header.css:62]: I may be wrong but I think the url can also be written like `/static/image/carousel-bg-1.png`

File: src/Components/Header/Header.tsx [hunk 8]
```diff
@@ -1,13 +1,109 @@
...
+                            <Modal
+                                show={show}
+                                onHide={handleClose}
+                                backdrop="static"
+                                keyboard={false}
+                            >
+                                <Modal.Header closeButton>
+                                    <Modal.Title>Current Location</Modal.Title>
+                                </Modal.Header>
+                                <Modal.Body>have to implement</Modal.Body>
```
Comment (Reeju2019) [inline src/Components/Header/Header.tsx:51]: Make the data dynamic
`<Modal.Body>{data.modal.body}</Modal.Body>`

File: src/Components/Header/Header.tsx [hunk 9]
```diff
@@ -1,13 +1,109 @@
...
+                                        Understood
+                                    </Button>
+                                </Modal.Footer>
+                            </Modal>
+
+                            {/* <!--End of Modal -->  */}
+
+                            <InputGroup className="search_btn d-flex">
+                                <div className="align-self-center m-3">
+                                    <i className="fas fa-search"></i>
```
Comment (Reeju2019) [inline src/Components/Header/Header.tsx:69]: It can also be written as
` <i className="fas fa-search"/>`

File: src/Components/Header/Header.tsx [hunk 10]
```diff
@@ -1,13 +1,109 @@
...
+                                href="home"
+                                className="text-muted fw-bold"
+                            >
+                                Home
+                            </Nav.Link>
+                            <Nav.Link
+                                href="contact"
+                                className="text-muted fw-bold"
+                            >
+                                Contact Us
```
Comment (Reeju2019) [inline src/Components/Header/Header.tsx:92]: Refer to the other comment and do the required changes

File: src/Components/Header/Header.tsx [hunk 11]
```diff
@@ -1,13 +1,109 @@
...
+                            </Nav.Link>
+                            <Nav.Link
+                                href="/login"
+                                className="d-flex flex-row text-muted fw-bold"
+                            >
+                                <i className="fa-regular fa-user"></i>
+                                Login
```
Comment (Reeju2019) [inline src/Components/Header/Header.tsx:99]: Refer to the other comments

File: src/Components/Header/Header.tsx [hunk 12]
```diff
@@ -1,13 +1,109 @@
...
+                                href="contact"
+                                className="text-muted fw-bold"
+                            >
+                                Contact Us
+                            </Nav.Link>
+                            <Nav.Link
+                                href="/login"
+                                className="d-flex flex-row text-muted fw-bold"
+                            >
+                                <i className="fa-regular fa-user"></i>
```
Comment (Reeju2019) [inline src/Components/Header/Header.tsx:98]: I have shared the code snippet for this inside another comment

File: src/Components/Header/HeaderCarousel.tsx [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+                                        src="/static/image/carousel-img-1.png"
+                                        alt="First slide"
+                                    />
+                                </div>
+                            </Carousel.Item>
+                            <Carousel.Item>
+                                <div className="d-flex flex-row">
+                                    <div className="w-100 d-flex justify-content-center align-item-center align-self-center">
+                                        <h2 className="txt col-8 justify-content-start text-white fs-2">
+                                            {HeaderData.header_caro_2_1} <br />{' '}
```
Comment (Reeju2019) [inline src/Components/Header/HeaderCarousel.tsx:36]: After br `{' '}` is this really needed?

**Review by Reeju2019 — CHANGES_REQUESTED** [review]

File: src/Components/Header/Header.tsx (hunk 9 above)
Comment (Mihir0000) [reply src/Components/Header/Header.tsx:69]: done

File: src/Components/Header/Header.tsx (hunk 11 above)
Comment (Mihir0000) [reply src/Components/Header/Header.tsx:99]: done

File: src/Components/Header/Header.tsx (hunk 12 above)
Comment (Mihir0000) [reply src/Components/Header/Header.tsx:98]: done

File: src/Components/Header/Header.tsx (hunk 8 above)
Comment (Mihir0000) [reply src/Components/Header/Header.tsx]: done

File: src/Components/Header/Header.tsx (hunk 10 above)
Comment (Mihir0000) [reply src/Components/Header/Header.tsx]: done

File: src/Components/Header/Header.css (hunk 7 above)
Comment (Mihir0000) [reply src/Components/Header/Header.css]: can't fix with `/static/image/carousel-bg-1.png`

**Pull request merged**